packages/utils/parcel-watcher-watchman-js/src/wrapper.js

```diff
@@ -41,26 +42,73 @@ export interface Watcher {
 export class ParcelWatcherWatchmanJS implements Watcher {
   subscriptionName: string;
   client: watchman.Client;
+  initPromise: Promise<void> | void;
 
   constructor() {
     this.subscriptionName = 'parcel-watcher-subscription-' + Date.now();
     this.client = new watchman.Client();
   }
 
-  commandAsync(args: any[]): Promise<any> {
+  init(watchDir: string): Promise<void> {
+    if (this.initPromise) {
+      return this.initPromise;
+    }
+
+    const client = this.client;
+
+    this.initPromise = new Promise((resolve, reject) => {
+      client.capabilityCheck({optional: [], required: []}, error => {
+        if (error) {
+          client.end();
+          reject(error);
+          return;
+        }
+
+        // Initiate the watch
+        client.command(['watch-project', watchDir], (error, resp) => {
+          if (error) {
+            client.end();
+            reject(error);
+            return;
+          }
+
+          // $FlowFixMe This code is straight from the watchman docs: https://facebook.github.io/watchman/docs/nodejs#initiating-a-watch
+          if ('warning' in resp) {
+            // $FlowFixMe
+            logger.warn({message: resp.warning});
+          }
+
+          resolve();
+        });
+      });
+    });
+
+    return this.initPromise;
+  }
+
+  async commandAsync(
+    command: string,
+    watchDir: string,
```

Comment:
Not all watchman commands receive these two parameters ; you can just make it variadic if you don't want the array wrapper on the calls

Reply:
Either keep as it was or make `args` variadic (`...args: `)

Reply:
I changed it because we need access to watchdir to ensure things are initialised. All the commands we actually use in the client follow this pattern. I can change though if you prefer.

Reply:
Understand. Is this what the C++ wrapper does? It spawns `watch ...` for whatever directory it got commands for?

Reply:
I guess so. This is just how watchman works.